### ISSUE TITLE
After load_weight loss become huge

### ISSUE THREAD

**Issue by Zjhzzzzz** [zylo117/Yet-Another-EfficientDet-Pytorch#169]
when I suspend the process of train and want to continue training, the loss of train become huge, like that my model don't have been trained.
I didn't change the load code
`model.load_state_dict(torch.load(weights_path), strict=False)`
I am confused of this phenomenon, can anyone help me?
Thanxxxxxxxxxxx!

**Comment by zylo117** [comment]
did you use `-w last`?

**Comment by Zjhzzzzz** [comment]
> did you use `--load_weights last`?

Yes, I did do this, I use`python train.py --load_weights logs/.../...pth`

**Comment by zylo117** [comment]
is there any warning? I couldn't replicate this error. Loading previous weights is ok here.

**Comment by Zjhzzzzz** [comment]
> is there any warning? I couldn't replicate this error. Loading previous weights is ok here. Try using `--load_weights last` instead

I tried use last instead, but still cannot solve my problem, cls_loss and reg_loss all behave like my model don't have been trained.
There is no warning message when I load weight.
I use coco_eval.py to see the result of different weights eval, I sadly find that all the result are same. Does it mean my train weights didn't save?

**Comment by Zjhzzzzz** [comment]
> is there any warning? I couldn't replicate this error. Loading previous weights is ok here. Try using `--load_weights last` instead

I didn't use COCO dataset, I use [DOTA2COCO](https://github.com/CAPTAIN-WHU/DOTA_devkit/blob/master/DOTA2COCO.py) to transform DOTA dataset to be COCO-like, maybe the problem exists here?

**Comment by zylo117** [comment]
you need to manully remove val_xxxx.json before evaluation, if you are not using the latest code.
Maybe you can pull the latest code and try again.

**Comment by Zjhzzzzz** [comment]
> you need to manully remove val_xxxx.json before evaluation, if you are not using the latest code.
> Maybe you can pull the latest code and try again.

I get result A(after 100 epochs with no weights load) is much better than result B(after 10 epochs with load xxxxd0_100_xx.pth weights ).Maybe It means that I can save weights but can't load weights. I try to change `torch.load(weights_path)`  to `torch.load(weights_path, map_location=xxx)` like coco_eval. But still cannot solve this problem.

What should I do if I want to solve this problem or where is the problem appear.

**Comment by zylo117** [comment]
Is this the latest code? Can you provide the mininum code to replicate this issue?

**Comment by Zjhzzzzz** [comment]
> Is this the latest code? Can you provide the mininum code to replicate this issue?

The code is downloaded about a week ago. I use `print(ret)` find that it ret is `<All keys matched successfully>`.
I just changed the code about project and didn't change anything.

**Comment by Zjhzzzzz** [comment]
> Is this the latest code? Can you provide the mininum code to replicate this issue?

My coco_eval load can success but train load fail.

**Comment by Zjhzzzzz** [comment]
> Is this the latest code? Can you provide the mininum code to replicate this issue?

Maybe problem don't exists in load but in train process?

**Comment by zylo117** [comment]
It might, but I can't replicate this issue. I tried to interrupt the training during the tutorial training, and then load it back with `--load_weights last`, the loss continues.

**Comment by zylo117** [comment]
Is it possible that it fails to save the weights? Can you load the last weights and infer a image to get a normal result?

**Comment by Zjhzzzzz** [comment]
> Is it possible that it fails to save the weights? Can you load the last weights and infer a image to get a normal result?

I use coco_eval.py and efficientdet_test.py to test whether I safe the weight. And the result show that coco_eval can get a normal eval and efficientdet_test can test the img and draw some box.
Both of them can load weights successfully, but train cannot.

**Comment by Zjhzzzzz** [comment]
> Is it possible that it fails to save the weights? Can you load the last weights and infer a image to get a normal result?

Is the problem might in model.train()? in model.eval() the model can load weights.

**Comment by zylo117** [comment]
you should pull the latest code and try again, it might have fixed it by now

**Comment by Zjhzzzzz** [comment]
> you should pull the latest code and try again, it might have fixed it by now

I have pull the latest code but the loss after load_weights is still huge like intial.

**Comment by zylo117** [comment]
Can you provide the mininum code to replicate this issue?

**Comment by Zjhzzzzz** [comment]
I just change --project to another datasets and don't change the code you pull in this project.
I don't know whether in coco datasets it will be replicated.
here is my DOTA.yml

`project_name: DOTA  # also the folder name of the dataset that under data_path folder
train_set: train
val_set: val
num_gpus: 1

mean: [0.485, 0.456, 0.406]
std: [0.229, 0.224, 0.225]

# this is coco anchors, change it if necessary
anchors_scales: '[2 ** 0, 2 ** (1.0 / 3.0), 2 ** (2.0 / 3.0)]'
anchors_ratios: '[(1.0, 1.0), (1.4, 0.7), (0.7, 1.4)]'

obj_list: ['plane', 'baseball-diamond',
           'bridge', 'ground-track-field',
           'small-vehicle', 'large-vehicle',
           'ship', 'tennis-court',
           'basketball-court', 'storage-tank',
           'soccer-ball-field', 'roundabout',
           'harbor', 'swimming-pool',
           'helicopter']`

**Comment by zylo117** [comment]
It is weird. I think you should clean up your environment.
Because I just did it again on colab. It worked just fine, and the loss continued.

Check this out.
I stopped the training process.
![Screenshot from 2020-04-23 16-57-18](https://user-images.githubusercontent.com/5842006/80080234-e9539380-8583-11ea-8dda-bc88dad277b6.png)

Then I resumed the training process.
![Screenshot from 2020-04-23 16-58-17](https://user-images.githubusercontent.com/5842006/80080282-fb353680-8583-11ea-9b64-eef7f381623d.png)

You can try that on colab or other servers/ computers, it should work.

**Comment by Zjhzzzzz** [comment]
> weird

ok, thank you, I will try.

**Comment by xungeer29** [comment]
我在其他项目上也遇到过这种问题，train加载之前的模型损失很大，但是inference没问题，我觉得是我哪里写错了，但是没找到错误之处
